pytrip/vdx.py

```diff
@@ -204,12 +212,45 @@ def voi_names(self):
         names = [voi.name for voi in self.vois]
         return names
 
-    def add_voi(self, voi):
+    def add_voi(self, voi, added_manually=False):
         """ Appends a new voi to this class.
+        If added_manually flag is True, sets voi color as one from spare ones.
+        If there is no spare colors, calls assign_voi_colors
 
         :param Voi voi: the voi to be appended to this class.
+        :param bool added_manually: flag that changes way VOI is added, default False
         """
-        self.vois.append(voi)
+        if added_manually:
+            if len(self._spare_voi_colors) > 0:
+                color = self._spare_voi_colors.pop()
+                voi.set_color(color)
+                self.vois.append(voi)
+            else:
+                self.assign_voi_colors()
+        else:
+            self.vois.append(voi)
+
+    def assign_voi_colors(self, k=3):
+        """
+        Creates n+k colors, where n is length of VOI list and k is size of a buffer for VOIs added in runtime.
```

Comment:
This method not only creates colors, but modifies list of Voi's stored in this class and applies colors to them. This behaviour is not mentioned in the docstring

Reply:
Right, I have had it in mind - that is why method is called as it is.

Reply:
Done